### PR TITLE
lib/scanner: Correctly scan symlinks (#3445)

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

Scan symlinks. :(

### Testing

Unit test.

### CODE REVIEW

**Comment by xduugu** [comment]
This seems to work. The Info messages disappeared and the symlinks are still there on the receiving side.

By the way, don't you need a keyword like "fixes" in the commit message to close the issue?

**Comment by calmh** [comment]
Indeed I do, fixed. :)

**Comment by calmh** [comment]
@st-review merge it, xduugu tested it

**Comment by st-review** [comment]
@calmh: Build status is `pending`. I'll wait until it goes green and then merge!

File: lib/scanner/walk_test.go [hunk 1]
```diff
@@ -281,6 +281,56 @@ func TestIssue1507(t *testing.T) {
...
 
+func TestWalkSymlink(t *testing.T) {
+	if !symlinks.Supported {
+		t.Skip("skipping unsupported symlink test")
+		return
+	}
+
+	// Create a folder with a symlink in it
+
+	os.RemoveAll("_symlinks")
```
Comment (AudriusButkevicius) [inline lib/scanner/walk_test.go]: osutil, because this will be flaky on windows I suspect.

Comment (calmh) [inline lib/scanner/walk_test.go]: inb4 merge

**Comment by xduugu** [comment]
Thanks for the quick fix @calmh

**Comment by st-review** [comment]
:ok_hand: Merged as 66a506e72b9dcc749d09a03cb120ba86bbf3d7f8. Thanks, @calmh!

**Pull request closed without merging**